tests/xtimer_now32_overflow/main.c

```diff
@@ -1,91 +1,59 @@
 /*
- * Copyright (C) 2020 Freie Universität Berlin
+ * Copyright (C) 2020 Kaspar Schleiser <kaspar@schleiser.de>
  *
  * This file is subject to the terms and conditions of the GNU Lesser
  * General Public License v2.1. See the file LICENSE in the top level
  * directory for more details.
  */
 
 /**
+ * @ingroup     tests
  * @{
  *
  * @file
- * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ * @brief       timer test application
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
  */
 
 #include <stdio.h>
 
-#include "fmt.h"
-#include "kernel_defines.h"
-#include "msg.h"
-#include "test_utils/expect.h"
-#include "thread.h"
 #include "xtimer.h"
 
-#define MAIN_MSG_QUEUE_SIZE     (4U)
-#define TIMERS_NUMOF            (3U)
-
-msg_t _main_msg_queue[MAIN_MSG_QUEUE_SIZE];
-static const uint64_t _timers_offsets[TIMERS_NUMOF] = {
-    /* MUST ASCEND */
-    1 * US_PER_SEC,
-    2 * US_PER_SEC,
-    3 * US_PER_SEC,
-};
+static void _callback(void *arg)
+{
+    (void)arg;
+}
 
 int main(void)
 {
-    xtimer_t timers[TIMERS_NUMOF];
-    msg_t msgs[TIMERS_NUMOF];
-    uint64_t start;
+    xtimer_t t1 = { .callback=_callback };
+    xtimer_t t2 = { .callback=_callback };
 
-    expect(ARRAY_SIZE(_timers_offsets) == TIMERS_NUMOF);
-    msg_init_queue(_main_msg_queue, MAIN_MSG_QUEUE_SIZE);
-    /* ensure that xtimer_now64() is greater than UINT32_MAX */
-    _xtimer_current_time = (2LLU << 32U);
+    /* ensure that xtimer_now64() is greater than UINT32_MAX
+     * and the upper 32bit of xtimer_now64() equal 1 */
+    _xtimer_current_time = (1LLU << 32U);
     xtimer_init();
-    print_str("Setting ");
-    print_u32_dec(TIMERS_NUMOF);
-    print_str(" timers:\n");
-    for (unsigned i = 0; i < TIMERS_NUMOF; i++) {
-        msgs[i].content.value = i;
-        print_str(" #");
-        print_u32_dec(i);
-        print_str(" in ");
-        print_u64_dec(_timers_offsets[i]);
-        print_str(" usec\n");
+
+    /* set to 100s (far in the future) */
+    xtimer_set(&t1, 100000000LU);
+    xtimer_set(&t2, 100000000LU);
```

Comment:
Why set two timers if only one is checked?

Reply:
The comment below kind of explains it, but not very well, if you don't understand what the problem is.

Reply:
Got it!